### PR TITLE
Added test for group and user commands for lx images

### PR DESCRIPTION
**Pull request by askfongjojo**
Hi @chorrell,

This is an example of the type of tests I'd like to start getting more coverage in LX images. They are not application-specific but could break many things when they have regressed after certain bug fixes or refactoring work. They are also not covered in the java/perl/ltp language regression test suite AFAIK.

If the format/breakdown looks fine to you, I'll work on some more basic test cases (that have no external dependencies). After that, I'll move on to NFS commands and discuss more about the setup with you when I get to that point.

Thanks,
Angela

### CODE REVIEW

**Comment by chorrell** [comment]
Looks good!

I think we could combine the group_spec.rb and user_spec.rb tests and put them in common for use with all the various distros. We could use some conditional blocks for where CentOS and Debian/Ubuntu diverge,

**Comment by chorrell** [comment]
Actually, we could always do that kind of refactoring later.

**Comment by askfongjojo** [comment]
Thanks @chorrell . It should be an easy change. I'll rework it and send another PR, and make sure i don't forget about comments to briefly describe the tests.

File: spec/debian/user_spec.rb [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+describe command('chown -R bar /opt/bar') do
+  its(:exit_status) { should eq 0 }
+end
+
+describe file('/opt/bar/test.sh') do
+  it { should exist }
+  it { should be_writable.by_user('bar') }
+end
+
+describe command('rm -R /opt/bar && rm -R /home/bar') do
```
Comment (chorrell) [inline spec/debian/user_spec.rb]: For consistency, maybe change this command to:

```
rm -Rf /opt/bar && rm -Rf /home/bar
```

File: spec/ubuntu/user_spec.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+describe command('chown -R bar /opt/bar') do
+  its(:exit_status) { should eq 0 }
+end
+
+describe file('/opt/bar/test.sh') do
+  it { should exist }
+  it { should be_writable.by_user('bar') }
+end
+
+describe command('rm -R /opt/bar && rm -R /home/bar') do
```
Comment (chorrell) [inline spec/ubuntu/user_spec.rb]: For consistency, maybe change this command to:

```
rm -Rf /opt/bar && rm -Rf /home/bar
```

**Comment by chorrell** [comment]
Sounds good!

**Pull request closed without merging**